app/models/pxe_server.rb

```diff
@@ -198,6 +198,14 @@ def delete_provisioning_files(pxe_image, mac_address, windows_image = nil, custo
     _log.info("#{log_message}...Complete")
   end
 
+  def ensure_menu_list(menu_list)
+    current_menus = pxe_menus.map(&:file_name)
+    to_destroy = current_menus.reject { |menu| menu_list.include?(menu) }
+    to_create = menu_list.reject { |menu| current_menus.include?(menu) }
```

Comment:
I believe this can be simplified to 

```ruby
current_menus = pxe_menus.map(&:file_name)
to_destroy = current_menus - menu_list
to_create = menu_list - current_menus
```